Added troubleshooting guide (#338)
GetStream/stream-video-swift: docusaurus/docs/iOS/06-advanced/04-troubleshooting-calls.mdx

---
title: Troubleshooting Guide
---

There are several possible integration issues that can lead to calls not being established. This section will cover the most frequent ones.

### Connection issues

> Connection issues usually happen when you provide an invalid token when setting up the SDK. When this happens, a web socket connection can't be established with our backend, resulting in errors when trying to connect to a call.

Reviewer: ```suggestion
Connection issues usually happen when you provide an invalid token during the SDK setup. When this happens, a web socket connection can't be established with our backend, resulting in errors when trying to connect to a call.
```

As you would do it in the real world, you would need to specify another member (or members) that you want to call. Another important note - that member should also exist in Stream's platform (it must have connected at least once). This is needed because we need to know the user's device and where to send the call notification.

#### Reusing a call id

Call IDs in general can be reused - you can join a call with the same id many times. However, the ringing is done only once per call ID. Therefore, if you implement calls with ringing, make sure that you provide a unique ID every time, in order for the ring functionality to work. One option is to use a `UUID` as a call ID.

#### CallKit integration issues

If you followed the CallKit [guide](https://getstream.io/video/docs/ios/advanced/callkit-integration/), and still have issues, here are some troubleshooting steps:
- make sure there are no connection issues (see points above)
- check if the generated VoIP certificate matches the bundle id specified in the dashboard
- check if the app is using the correct bundle id

> - check if have created push providers and you specified their correct names when creating the SDK

Reviewer: ```suggestion
- check if you have created push providers and you specified their correct names when creating the SDK
```

> - check if you registered the notification device correctly, by examining whether it's returned in the `me` response in the `connection.ok` event

Reviewer: ```suggestion
- check if you registered the device correctly, by examining whether it's returned in the `me` response in the `connection.ok` event
```
